### PR TITLE
MMT-3552: Published record search

### PR DESCRIPTION
**Pull request by trevorlang**
This PR adds published searching for published collections, services, tools, and variables. Services, tools, and variables results are only available directly by URL until MMT-3493 makes them available via the search bar.

- A cmr-graphql `keyword` search is performed from the application header search bar
- The search results are displayed on the `/search` route
- Search parameters are pulled from the URL parameters
- Results are set to 20 per page
- Pagination will be displayed when multiple pages exist
- Changes were made to the `Table` component to make it a little easier to configure
- Columns can be set to enable sorting and will pass the `sortKey` to cmr-graphql

Search results page:
image

### CODE REVIEW

File: static/src/js/pages/SearchPage/SearchPage.jsx [hunk 1]
```diff
@@ -0,0 +1,450 @@
...
+      setSearchParams((currentParams) => {
+        currentParams.delete('sortKey')
+
+        return {
+          ...Object.fromEntries(currentParams)
+        }
+      })
+    }
+
+    if (!order) return
```
Comment (trevorlang) [inline static/src/js/pages/SearchPage/SearchPage.jsx:191]: Move this above

File: static/src/css/vendor/bootstrap/overrides/_table.scss [hunk 2]
```diff
@@ -3,24 +3,24 @@
...
 .table > thead > tr > th,
 .table > tbody > tr > td {
-  border: 0;
+  // border: 0;
 }
 
 .table > thead > tr > th,
 .table > tbody > tr > td {
-  border: 0;
+  // border: 0;
```
Comment (cgokey) [inline static/src/css/vendor/bootstrap/overrides/_table.scss:18]: remove comment

File: static/src/css/vendor/bootstrap/overrides/_table.scss [hunk 3]
```diff
@@ -3,24 +3,24 @@
...
-  padding: 0.75rem;
+  padding: 0.75rem 1rem;
 }
 
 .table > thead > tr,
 .table > tbody > tr,
 .table > thead > tr > th,
 .table > tbody > tr > td {
-  border: 0;
+  // border: 0;
```
Comment (cgokey) [inline static/src/css/vendor/bootstrap/overrides/_table.scss:13]: remove comment

File: static/src/css/vendor/bootstrap/overrides/_table.scss [hunk 4]
```diff
@@ -29,7 +29,7 @@
 
 .table > tbody > tr > td:first-child,
 .table > tbody > tr > td:last-child {
-  border-radius: 0.325rem;
+  // border-radius: 0.325rem;
```
Comment (cgokey) [inline static/src/css/vendor/bootstrap/overrides/_table.scss:32]: remove comment

File: static/src/css/vendor/bootstrap/overrides/_table.scss [hunk 5]
```diff
@@ -43,5 +43,5 @@
 }
 
 .table-bordered > :not(caption) > * > * {
-  border: 0;
+  // border: 0;
```
Comment (cgokey) [inline static/src/css/vendor/bootstrap/overrides/_table.scss:46]: remove comment

File: static/src/js/components/DraftList/DraftList.jsx [hunk 6]
```diff
@@ -59,63 +71,181 @@ const DraftList = ({ draftType }) => {
...
-      conceptId, revisionDate, ummMetadata
-    } = item
-    const {
-      ShortName, EntryTitle, Name, LongName
-    } = ummMetadata || {}
+  const setPage = (nextPage) => {
+    setActivePage(nextPage)
+  }
+
+  // // Building a Table using Data in items
```
Comment (cgokey) [inline static/src/js/components/DraftList/DraftList.jsx:78]: I guess  you can remove all this

File: static/src/js/components/DraftList/DraftList.jsx [hunk 7]
```diff
@@ -59,63 +71,181 @@ const DraftList = ({ draftType }) => {
...
+      </div>
+    )
+  }, [])
+
+  const [columns, setColumns] = useState([
+    {
+      dataKey: 'revisionDate',
+      title: 'Last Modified',
+      className: 'col-auto',
+      dataAccessorFn: (cellData) => cellData.split('T')[0]
```
Comment (cgokey) [inline static/src/js/components/DraftList/DraftList.jsx:192]: You may have gotten this name from react-table, but I think maybe cellRendererFn might be a little more descriptive?

File: static/src/js/components/DraftList/DraftList.jsx [hunk 8]
```diff
@@ -59,63 +71,181 @@ const DraftList = ({ draftType }) => {
...
+          dataAccessorFn: buildEllipsisTextCell
+        },
+        ...newColumns
+      ]
+    }
+
+    setColumns(newColumns)
+  }, [])
+
+  const currentPageIndex = Math.floor(offset / limit)
```
Comment (cgokey) [inline static/src/js/components/DraftList/DraftList.jsx:241]: I'm thinking this code will be repeated in multiple places, consider moving into a helper function where you just give it the offset, limit, and active page and will will return all these variables.  I see it is repeated in other places

**Review by dmistry1 — APPROVED** [review]

**Review by tle1989 — APPROVED** [review]

**Pull request merged**